<img src="./unal.png" align=left width="150" height="150"></img>

$\;$
---


<h2 align="center">PRUEBA DE MEDIAS (Poblaciones Normales y  $\sigma_1^2 = \sigma_2^2$ )
    <h3 align="center">"prueba dos poblaciones"

    
### Oscar Julian Layton 
---
    
$\;$



Para realizar una prueba de de media para dos muestras se tiene:

## Inferencia de medias para dos muestras

Considerese:

* $X_{11}, X_{12}, X_{13},...,X_{1 n_1}$ una _muestra aleatoria_ de una población distribuida Normal con media $\mu_1$ y varianza $\sigma^2_1$.
* $X_{21}, X_{22}, X_{23},...,X_{2 n_1}$ una _muestra aleatoria_ de una población distribuida Normal con media $\mu_2$ y varianza $\sigma^2_2$.
* $\sigma^2_1 = \sigma^2_2 = \sigma^2$
* Las muestras anteriores son independientes entre si.

$$E[\overline{X}_1 - \overline{X}_2 ]= E[\overline{X}_1] - E[\overline{X}_2] = \mu_1 - \mu_2$$

Ahora considerando que las muestras son independientes entre si y además  $\sigma_1^2 = \sigma_2^2 = \sigma^2$ :

$$V(\overline{X}_1 - \overline{X}_2 )=V(\overline{X}_1 ) + V(\overline{X}_2 ) = \frac{\sigma^2_1}{ n_1}  +  \frac{\sigma^2_2}{ n_2} = \sigma^2 \left (   \frac{1}{ n_1}  +  \frac{1}{ n_2}    \right )$$

En consecuencia:

* Como las distribuciones poblacionales son Normales, tanto $\overline{X}_1$ como $\overline{X}_2$ son normales, por lo tanto $ \overline{X}_1 - \overline{X}_2 $ es Normal, además $\sigma^2_1 = \sigma^2_2 = \sigma^2$

$$ \overline{X}_1 - \overline{X}_2  \;\; \sim \;\; N \left ( \mu_1 - \mu_2 \;\; ,\;\;   s_p \sqrt{  \frac{1}{ n_1}  +  \frac{1}{ n_2}    } \right ) $$

donde:
$$s_p^2=\frac{(n_1 - 1)s^2_1 + (n_2 - 1)s^2_2}{n_1 + n_2 -2}$$



#### 1. Parámetro Poblacional

El parámetro poblacional a contrastar es $\mu_1 = \mu_2$ "Igualdad de medias poblacionales"

#### 2. Hipótesis

Considérese las medias poblacionales $\mu_1$  $\mu_2$


\begin{equation}
H_{0}: \mu_1 - \mu_2 = \Delta_o \hspace{0.9cm} vs \hspace{0.9cm} H_{1}: \mu_1 - \mu_2 > \Delta_o
\end{equation}

#### 3. Estadística de prueba

<div class="alert alert-success">
Supóngase que $ \{ x_{11}, x_{12}, ,..., x_{1n}  \}$ y $\{  x_{21}, x_{22}, ,..., x_{2n}  \}$son dos _muestras aleatorias_ independientes deonde $\sigma_1^2=\sigma_2^2$, luego

\begin{equation}
T=\frac{(\overline{X}_1-\overline{X}_2)-\Delta_o }{ s_p \sqrt{ \frac{1}{ n_1}  +  \frac{1}{ n_2}}}\hspace{0.4cm} \hspace{0.4cm}  \hspace{1.2cm} ;  \hspace{1.2cm} T\mid _{H_0} \sim t_{(n_1+n_2-2)}
\end{equation}

donde:
$$s_p^2=\frac{(n_1 - 1)s^2_1 + (n_2 - 1)s^2_2}{n_1 + n_2 -2}$$
    
</div>

#### 4. Construcción regla de desición

Para un $0<\alpha<1$ nivel de significancia

$$P_{\Delta_{o}}(Error\; tipo \; I)=\alpha$$
$$P_{\Delta_{o}}(Rechaz\;H_o\;cuando \;es \;verdadera)=\alpha$$
$$P_{\Delta_{o}}(T \geq t_{(\alpha \;,\;n_1+ n_2 -2  )} \; cuando \; T \sim t_{(n_1+n_2-2)}) =\alpha $$

En consecuencia:

$$P_{\Delta_{o}}\left (   \frac{(\overline{X}_1-\overline{X}_2)-\Delta_o }{s_p \sqrt{ \frac{1}{ n_1}  +  \frac{1}{ n_2}}}\geq t_{(\alpha \;,\;n_1+ n_2 -2)}   \right )=\alpha$$

#### 5. Regla de Desición


<div class="alert alert-danger">

Para un $\alpha$ de significancia se procede a:

\begin{equation}
\vartheta :  \; \;Rechazar \; H_o  \;\;\;\; sii \;\; \;\; T \geq t_{(\alpha \;,\;n_1+ n_2 -2)}
\end{equation}
</div>

---

Para la **potencia de la prueba** se usa el error tipo II

$$\beta(\mu')=P(Error\; tipo \; II) $$

$$\beta(\mu')= P(Z_c < t_{(\alpha \;,\;n_1+ n_2 -2)} \;\; cuando \; \;\mu_1 - \mu_2 =\Delta^{´})$$


$$\beta(\mu') = P \left (   \overline{X}_1 -\overline{X}_2  < \Delta_o + t_{(\alpha \;,\;n_1+ n_2 -2)} s_p\left (  \sqrt{\frac{1}{ n_1}  +  \frac{1}{ n_2}}  \right )  \;\; con \;\; \mu_1 - \mu_2 =\Delta^{´}   \right )$$

$$\beta(\mu') =  \left (    \frac{(\overline{X}_1-\overline{X}_2)-\Delta^{´} }{s_p \sqrt{ \frac{1}{ n_1}  +  \frac{1}{ n_2}} } \; < \; t_{(\alpha \;,\;n_1+ n_2 -2)} + \frac{\Delta_o- \Delta^{´}}{s_p \sqrt{ \frac{1}{ n_1}  +  \frac{1}{ n_2}}  } \right )$$
$$potencia = 1 - \beta$$

----
## Ejemplo

Se investigan tipos de aguacates, un agricultor midió los pesos de 17 los aguacates más maduros, los cuales pertenecían a dos tipos de especies diferentes: Criollo (c) y Hass (s). El Agricultor cree que los frutos que están en el piso -maduros  de tipo **Criollo** tienen en promedio un peso mayor que el peso promedio de los aguacates tipo **Hass**. Se desea comprobar estadísticamente la afirmación del agricultor. Los datos experimentales obtenidos por el investigador se presentan en la siguiente tabla:

|Tipo|Peso| <div style="width:50px"> </div>|Tipo|Peso|
|---|---|---|---|---|
|h |15.16 | |h| 30.86|
|c |28.99 | |h| 14.00|
|c |16.95 | |h| 38.72|
|c |29.61 | |h| 23.29|
|c |35.61 | |h| 17.00|
|c |20.69 | |h| 35.40|
|c |17.39 | |h| 19.32|
|c |29.94 | |h| 29.88|
|c |22.79 | | |      |

In [3]:
peso <- c(15.16,28.99,16.99,29.61,35.61,20.69,17.39,29.94,22.79,30.86,14.00,
          38.72,23.29,17.00,35.40,19.32,29.88)
especie <- c('h',rep('c',8),rep('h',8))
data <- data.frame(especie,peso)

In [7]:
c<-data$peso[data$especie=='c']
h<-data$peso[data$especie=='h']

#### Paso 1) Prueba varianzas poblacionales. (primer paso para pruebas de dos poblaciones)

Como primera medida se debe conocer si $\sigma^2_h = \sigma^2_c$, la prueba se encuentra en el script [Prueba para igualdad de varianzas poblacionales.](./44_prueba_hipotesis_varianza_dos_poblaciones.ipynb). De donde se concluye que $\sigma_h^2 = \sigma_c^2$

### b) Hipótesis sobre medias poblacionales.

Como primera medida se debe contrastar la hiótesis:

\begin{equation}
H_{0}: \mu_1 - \mu_2 = \Delta_o \hspace{0.9cm} vs \hspace{0.9cm} H_{1}: \mu_1 - \mu_2 > \Delta_o
\end{equation}

que es  lo mismo que:

\begin{equation}
H_{0}: \mu_c - \mu_h = \Delta_o \hspace{0.9cm} vs \hspace{0.9cm} H_{1}: \mu_c - \mu_h > \Delta_o
\end{equation}

### b) Uso de la estadística de prueba.

Se tiene el siguiente Estadístico de prueba donde $\Delta_0 =0$:

\begin{equation}
T=\frac{(\overline{X}_1-\overline{X}_2)-\Delta_o }{ s_p \sqrt{ \frac{1}{ n_1}  +  \frac{1}{ n_2}}}\hspace{0.4cm} \hspace{0.4cm}  \hspace{1.2cm} ;  \hspace{1.2cm} T\mid _{H_0} \sim t_{(n_1+n_2-2)}
\end{equation}

donde:
$$s_p^2=\frac{(n_1 - 1)s^2_1 + (n_2 - 1)s^2_2}{n_1 + n_2 -2}=66.03$$

verifiquemos:

In [37]:
n1 <-8
n2 <-9

s_p_cuadrado <- (1/(n1 + n2 -2))*((n1 -1)*var(c)+(n2 -1)*var(h)) ; 
s_p<-sqrt(s_p_cuadrado)  ; s_p 

[1] 8.126218

In [38]:
mean1 <- mean(c)
mean2 <- mean(h) 

In [40]:
T<-(mean1-mean2)/(s_p*sqrt(1/n1 + 1/n2 ))  ; T

[1] 0.1021803

En consecuencia:

In [57]:
##########################  cuantil distribucion t ################################################################
######     distribución t     qt(p, df, ncp, lower.tail = TRUE, log.p = FALSE)  #si es central npc=0  #############  
###################################################################################################################

alpha=0.05
df=n1+n2-2    #por ser dos muestras 

qt(alpha/2,df, 0, lower.tail = FALSE)

[1] 2.13145

#### c) Desición

 $ \vartheta :$ Luego como $T= 0.102$, se tiene que a un nivel de significancia de $ \alpha= 0.05 $, $\;\;$ $T= 0.102 < t_{[15\;,\;0.05/2]}=0.025 $ con lo cual se concluye que NO hay evidencia estadísticamente significativa para rechazar la hipotesis nula, luego no puede aceptarse la afirmación del agricultor. esto tambien se evidencia en el p_valor (el cual es muy grande):


In [62]:
df=n1+n2-2 
pt(T, df, 0, lower.tail = FALSE)   #lower.tail=TRUE ent P[X >= x] 

[1] 0.4599835

---
#### d) Prueba en R junto con intervalos de confianza

In [63]:
# Prueba de igualdad de medias #
t.test(peso~especie,var.equal=T,alternative="greater")


	Two Sample t-test

data:  peso by especie
t = 0.10218, df = 15, p-value = 0.46
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 -6.518679       Inf
sample estimates:
mean in group c mean in group h 
       25.25125        24.84778 
